lumicks/pylake/kymotracker/detail/trace_line_2d.py

```diff
@@ -180,14 +182,23 @@ def _traverse_line_direction(
     return nodes
 
 
+@dataclass(frozen=True)
 class KymoLineData:
-    __slots__ = ["time_idx", "coordinate_idx"]
+    """Data class which holds the raw data for a Kymograph line.
+
+    Parameters
+    ----------
+    time_idx : List[float]
+        List of time indices on the tracked image [pixels]
+    coordinate_idx : List[float]
+        List of coordinate indices on the tracked image [pixels]
+
+    Note: Some of the kymotracker functions return numpy arrays. Appending too many small numpy
+    arrays was a bottleneck and therefore we explicitly use lists."""
 
-    def __init__(self, time_idx, coordinate_idx):
-        """Some of the kymotracker classes return numpy arrays. Appending to many small numpy
-        arrays was a bottleneck and therefore we explicitly use lists."""
-        self.time_idx = list(time_idx)
-        self.coordinate_idx = list(coordinate_idx)
+    __slots__ = ["time_idx", "coordinate_idx"]
```

Comment:
This may be a weird git rebase leftover. This is not needed any more with dataclasses.